controllers/cloudstackmachine_controller.go

```diff
@@ -186,6 +187,7 @@ func (r *CloudStackMachineReconciliationRunner) DeleteMachineIfFailuredomainNotE
 // Implicitly it also fetches its bootstrap secret in order to create said instance.
 func (r *CloudStackMachineReconciliationRunner) GetOrCreateVMInstance() (retRes ctrl.Result, reterr error) {
 	if r.CAPIMachine.Spec.Bootstrap.DataSecretName == nil {
+		r.Recorder.Event(r.ReconciliationSubject, "Normal", "Creating", "Bootstrap DataSecretName not yet available.")
```

Comment:
why not write to the CAPI machine?

Reply:
Tried in my local, technically, we can create an event and bind it to CAPI machine. 
After discussed with team, we concluded that putting events under CAPC only was enough right now. 

Thanks for the idea.